typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,541 @@
+                if (error) {
+                    resolve(errAsync(error));
+                } else {
+                    resolve(okAsync(response));
+                }
+            })
+        });
+        const resultRegisterStaticFiles: Result<RegisterStaticFilesResponse, Error> = await promiseRegisterStaticFiles;
+        if (!resultRegisterStaticFiles.isOk()) {
+            return err(resultRegisterStaticFiles.error);
+        }
+        const resp: RegisterStaticFilesResponse = resultRegisterStaticFiles.value;
+
+        const staticFileDestRelativeFilepathsMap: Map<string, string> = resp.getStaticFileDestRelativeFilepathsMap();
+        for (let [staticFileIdStr, destFilepathRelativeToEnclaveVolRoot] of staticFileDestRelativeFilepathsMap.entries()) {
+
+            const staticFileId: StaticFileID = <StaticFileID>(staticFileIdStr);
+
+            if (!staticFileFilepaths.has(staticFileId)) {
+                return err(new Error("No source filepath found for static file " + staticFileId + "; this is a bug in Kurtosis"));
+            }
+            const srcAbsFilepath: string = staticFileFilepaths[staticFileId];
+
+            const destAbsFilepath: string = path.join(this.enclaveDataVolMountpoint, destFilepathRelativeToEnclaveVolRoot);
+            
+            const promiseDestAbsFilepath: Promise<fs.Stats> = fsPromises.stat(destAbsFilepath);
+            promiseDestAbsFilepath.catch(() => { 
+                return err(new Error("The Kurtosis API asked us to copy static file " + staticFileId + " to path " + destFilepathRelativeToEnclaveVolRoot + 
+                " in the enclave volume which means that an empty file should exist there, " + "but no file exists at that path - this is a bug in Kurtosis!"));
+            })
+
+            
+            var srcFp: number;
+            var destFp: number;
+            try { //TODO (comment) - finally block meant to duplicate defer functionality
+                
+                const promiseSrcFp: Promise<fsPromises.FileHandle> = fsPromises.open(srcAbsFilepath, 'r');
+                promiseSrcFp.catch(error => { 
+                    return err(error); 
+                })
+                const resultSrcFp: fsPromises.FileHandle = await promiseSrcFp;
+                srcFp = resultSrcFp.fd;
+                
+                const promiseDestFp: Promise<fsPromises.FileHandle> = fsPromises.open(destAbsFilepath, 'w');
+                promiseDestFp.catch(error => { 
+                    return err(error); 
+                })
+                const resultDestFp: fsPromises.FileHandle = await promiseSrcFp;
+                destFp = resultDestFp.fd;
+    
+                const promiseCopyFile: Promise<void> = fsPromises.copyFile(srcAbsFilepath, destAbsFilepath);
+                promiseCopyFile.catch(error => { 
+                    return err(error); 
+                })
+            }   
+            finally {
+                fs.close(srcFp);
```

Comment:
There's actually a subtle bug with the way this is being done. Let's say this line threw an error:
```
promiseSrcFp.catch(error => {
```
Then, the `finally` block would run and BOTH `srcFp` and `destFp` would be null, which would lead to two NPEs here. The `defer` location of the Go code was chosen very carefully (i.e., RIGHT after the error-checking, since if an error is returned then the result `fp` will be null and can't have `Close()` called on it) and you'll need to duplicate that here. This will look like something like this (make sure to clean this up if you copy-paste directly):
```
openSrcFpPromise = ......creating promise......
openSrcFpResult = await openSrcFpPromise
if !openSrcFpRersult.isOk() {
    return err(......something.....)
}
srcFp = openSrcFpResult.value
try {
    openDestFpPromise = ......creating promise......
    openDestFpResult = await openDestFpPromise
    if !openDestFpRersult.isOk() {
        return err(......something.....)
    }
    destFp = openDestFpResult.value
    try {
         ......stuff......
    } finally {
         fp.close(destFp)
    }
} finally {
    fp.Close(openSrcFp)
}
```
Now you understand why Go's `defer`s are so nice ;)